toxcore/tox.h

```diff
@@ -3267,6 +3267,2167 @@ uint16_t tox_self_get_tcp_port(const Tox *tox, Tox_Err_Get_Port *error);
+     */
+    TOX_ERR_GROUP_SEND_PRIVATE_MESSAGE_PERMISSIONS,
+
+    /**
+     * Packet failed to send.
+     */
+    TOX_ERR_GROUP_SEND_PRIVATE_MESSAGE_FAIL_SEND,
+
+    /**
+     * The group is disconnected.
+     */
+    TOX_ERR_GROUP_SEND_PRIVATE_MESSAGE_DISCONNECTED,
+
+    /**
+     * The message type is invalid.
+     */
+    TOX_ERR_GROUP_SEND_PRIVATE_MESSAGE_BAD_TYPE,
+
+} Tox_Err_Group_Send_Private_Message;
+
+
+/**
+ * Send a text chat message to the specified peer in the specified group.
+ *
+ * This function creates a group private message packet and pushes it into the send
+ * queue.
+ *
+ * The message length may not exceed TOX_MAX_MESSAGE_LENGTH. Larger messages
+ * must be split by the client and sent as separate messages. Other clients can
+ * then reassemble the fragments. Messages may not be empty.
+ *
+ * @param group_number The group number of the group the message is intended for.
+ * @param peer_id The ID of the peer the message is intended for.
+ * @param message A non-NULL pointer to the first element of a byte array
+ *   containing the message text.
+ * @param length Length of the message to be sent.
+ *
+ * @return true on success.
+ */
+bool tox_group_send_private_message(const Tox *tox, uint32_t group_number, uint32_t peer_id, Tox_Message_Type type,
+                                    const uint8_t *message, size_t length, Tox_Err_Group_Send_Private_Message *error);
+
+typedef enum Tox_Err_Group_Send_Custom_Packet {
+
+    /**
+     * The function returned successfully.
+     */
+    TOX_ERR_GROUP_SEND_CUSTOM_PACKET_OK,
+
+    /**
+     * The group number passed did not designate a valid group.
+     */
+    TOX_ERR_GROUP_SEND_CUSTOM_PACKET_GROUP_NOT_FOUND,
+
+    /**
+     * Message length exceeded TOX_MAX_MESSAGE_LENGTH.
+     */
+    TOX_ERR_GROUP_SEND_CUSTOM_PACKET_TOO_LONG,
+
+    /**
+     * The message pointer is null or length is zero.
+     */
+    TOX_ERR_GROUP_SEND_CUSTOM_PACKET_EMPTY,
+
+    /**
+     * The caller does not have the required permissions to send group messages.
+     */
+    TOX_ERR_GROUP_SEND_CUSTOM_PACKET_PERMISSIONS,
+
+    /**
+     * The group is disconnected.
+     */
+    TOX_ERR_GROUP_SEND_CUSTOM_PACKET_DISCONNECTED,
+
+} Tox_Err_Group_Send_Custom_Packet;
+
+
+/**
+ * Send a custom packet to the group.
+ *
+ * If lossless is true the packet will be lossless. Lossless packet behaviour is comparable
+ * to TCP (reliability, arrive in order) but with packets instead of a stream.
+ *
+ * If lossless is false, the packet will be lossy. Lossy packets behave like UDP packets,
+ * meaning they might never reach the other side or might arrive more than once (if someone
+ * is messing with the connection) or might arrive in the wrong order.
+ *
+ * Unless latency is an issue or message reliability is not important, it is recommended that you use
+ * lossless packets.
+ *
+ * The message length may not exceed TOX_MAX_CUSTOM_PACKET_SIZE. Larger packets
```

Comment:
Thats probably how it's intended, but right now it checks for `MAX_GC_MESSAGE_SIZE`, which is 1 byte shorter.